build-parent/pom.xml

```diff
@@ -320,6 +323,24 @@
                 <version>${subethasmtp.version}</version>
                 <scope>test</scope>
             </dependency>
+
+            <!-- Artemis test dependencies -->
+            <dependency>
+                <groupId>org.apache.activemq</groupId>
+                <artifactId>artemis-server</artifactId>
+                <version>${artemis.version}</version>
+                <exclusions>
+                    <exclusion>
+                        <groupId>org.checkerframework</groupId>
+                        <artifactId>checker-qual</artifactId>
+                    </exclusion>
+                </exclusions>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.activemq</groupId>
+                <artifactId>artemis-amqp-protocol</artifactId>
+                <version>${artemis.version}</version>
+            </dependency>
```

Comment:
Are these test dependencies? If they are, I would suggest adding `test` scope. I know it's not a typical practice but it may help if one of the extensions start relying on these in their runtime and/or build classpath. Constraints configured in build-parent are managing the runtime and build classpaths but aren't present in the BOM users import. So we should be careful with these.

Reply:
Adjusted. I also added a bunch of exclusions.